src/openassetio-python/bridge/include/openassetio/python/converter.hpp

```diff
@@ -20,49 +20,51 @@ namespace python::converter {
  * This template is explicitly instantiated to only the OpenAssetIO
  * types, and is not intended to be a generic converter.
  *
- * The purpose of this function is allow Python/C++ conversion whilst
- * hiding the pybind11 dependency, allowing consumers to retrieve a
- * Python object without having to have pybind in their build stack. A
- * pybind cast is done behind the scenes, returning the released ptr
- * from that.
+ * The purpose of this function is to provide a Python/C++ conversion
+ * without requiring/exposing a particular CPython binding library
+ * implementation.
+ *
+ * The returned `PyObject` owns a hidden reference to the input
+ * `shared_ptr`, ensuring it is kept alive. The reference will be
+ * released when the `PyObject` is destroyed.
  *
  * @note This function does not acquire the GIL.
  *
  * @warning A Python environment, with openassetio imported, must be
  *          available in order to use this function.
  *
  * @throws std::invalid_argument if the input is null.
- * @throws std::runtime_error if the cast fails. The .what() of this
- *         exception is the Python exception string.
+ * @throws std::runtime_error if the cast fails.
  *
  * @param objectPtr An OpenAssetIO pointer type, (eg, @needsref
  * ManagerPtr). The returned pointer takes shared ownership of this
- * input \p objectPtr, and will keep the c++ instance alive until the
+ * input \p objectPtr, and will keep the C++ instance alive until the
  * `PyObject` is destroyed.
  *
- * @return A `PyObject` pointer to an object of the Python api type
- * associated with the c++ api object provided.
- * The pointer is not an RAII type, and will be returned with an
- * incremented reference count for the caller to manage,  and must
- * must be released via `Py_DECREF` or similar mechanisms.
+ * @return A `PyObject` pointer to an object of the Python API type
+ * associated with the C++ API object provided. The reference count of
+ * the `PyObject` will be incremented, and ownership of the reference
+ * transferred to the caller.
```

Comment:
[Consider] I'm unsure whether saying that the ownership is transferred is enough help for these docs. This is the only place in the API where you have to take manual ownership of a type and call a cleanup function, so I don't think assumed CPython knowledge should be taken for granted. I'd rather be more explicit and inform the caller that they need to call PY_DECREF.

Reply:
OK that's fair enough.  I updated this based on nomenclature in the [CPython docs](https://docs.python.org/3/extending/extending.html#ownership-rules).

Reply:
My thinking is that the context is different for us, in the CPython docs everything inherently must be DECREF'd so you don't mention it all the time, but here it's an exception, so we should mention it.